core/src/session.rs

```diff
@@ -77,6 +77,7 @@ struct SessionData {
     client_brand_name: String,
     client_model_name: String,
     connection_id: String,
+    auth_data: Vec<u8>,
```

Comment:
If (and only if) more ergonomical, you could consider using `Bytes` instead of `Vec<u8>`. I think we pull in that dependency anyway, and it may make lifetimes & ownership around this networking I/O easier.

Reply:
I have no experience using `Bytes`. But given someone might want to use this crate without the rest of librespot, maybe it's not worth forcing that extra dep on them?

Reply:
Is it actually a good idea to use `Bytes` here? Based on the code usage, the only usage of `auth_data()` method is in `get_token.rs` in examples which is used as `Credentials::auth_data` which is an owned `Vec<u8>`. If we were using `Bytes`, we had to create an owned `Vec<u8>` for it which I think defeats the purpose of the `Bytes` zero-copy clone method because of `Vec<u8>` allocation.
However, something else we can do is to change the `Credentials::auth_data` to be `Bytes` instead of `Vec<u8>`. I'm not sure what implications this change would have but we can investigate.
Also, if you are committed to use `Bytes` here, you can use these two functions:
```rust
    pub fn auth_data(&self) -> Bytes {
        self.0.data.read().auth_data.clone() // this is zero copy!
    }

    pub fn set_auth_data(&self, auth_data: &[u8]) {
        self.0.data.write().auth_data = Bytes::copy_from_slice(auth_data);
    }
```
Each `Bytes` can be converted to an `Vec<u8>` using [`to_vec`](https://docs.rs/bytes/1.7.1/bytes/struct.Bytes.html#method.to_vec) method. Also, based on the `Cargo.toml`, the `librespot-core` package contains the `Bytes` library as its dependency.

Reply:
Thanks for looking into this.

> Also, based on the Cargo.toml, the librespot-core package contains the Bytes library as its dependency.

The idea is that his new librespot-ouath crate can be used independently of librespot (e.g. someone's random code that uses Spotify's Web API) so the existing dependencies of librespot-core don't come into it.

Reply:
Like I said, no worries, only if it were more ergonomical.